File: app/src/main/java/com/zaf/econnecto/ui/adapters/StaggeredImageAdapter.java
package com.zaf.econnecto.ui.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;
import com.zaf.econnecto.R;
import com.zaf.econnecto.network_call.response_model.home.CategoryData;
import com.zaf.econnecto.network_call.response_model.img_data.ViewImageData;
import com.zaf.econnecto.ui.fragments.BizCategoryFragment;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class StaggeredImageAdapter extends RecyclerView.Adapter<StaggeredImageAdapter.ViewHolder> {

    private  List<ViewImageData> mValues;
    private Context context;

    public StaggeredImageAdapter(Context mContext, List<ViewImageData> items) {
        context = mContext;
        mValues = items;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.category_list_items, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.mItem = mValues.get(position);
        holder.textName.setText(mValues.get(position).getImgId().toString());
        //holder.textEstd.setText(mValues.get(position).getLastName());

        Picasso.get().load(mValues.get(position).getImageLink()).placeholder(R.drawable.default_biz_profile_pic).into(holder.imgItem);


    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final View mView;
        final TextView textName;
       // final TextView textEstd;
        final ImageView imgItem;
        ViewImageData mItem;

        ViewHolder(View view) {
            super(view);
            mView = view;
            textName = (TextView) view.findViewById(R.id.textName);
         //   textEstd = (TextView) view.findViewById(R.id.content);
            imgItem = (ImageView) view.findViewById(R.id.imgItem);
        }

        @NotNull
        @Override
        public String toString() {
            return super.toString() + " '" + textName.getText() + "'";
        }
    }
}

File: app/src/main/java/com/zaf/econnecto/ui/adapters/StaggeredImageAdapter.kt
package com.zaf.econnecto.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import com.zaf.econnecto.R
import com.zaf.econnecto.network_call.response_model.img_data.ViewImageData
import com.zaf.econnecto.ui.interfaces.DeleteImageListener


class StaggeredImageAdapter(private val context: Context, private
val mValues: MutableList<ViewImageData>, deleteVisible: Boolean, iEditImage: DeleteImageListener?)
    : RecyclerView.Adapter<StaggeredImageAdapter.ViewHolder>() {

    private val showDeleteIcon = deleteVisible
    private val editImageListener = iEditImage
    private val mContext: Context = context

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_photo_list_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mItem = mValues[position]
        holder.imgDelete.visibility = if (showDeleteIcon) View.VISIBLE else View.GONE
        Picasso.get().load(mValues[position].imageLink).placeholder(R.drawable.default_biz_profile_pic).into(holder.imgItem)
        holder.imgDelete.setOnClickListener {
            editImageListener?.onDeleteClick(holder.mItem, position)
        }
    }

    override fun getItemCount(): Int {
        return mValues.size
    }

    class ViewHolder internal constructor(private val mView: View) : RecyclerView.ViewHolder(mView) {
        val imgItem: ImageView = mView.findViewById<View>(R.id.imgItem) as ImageView
        val imgDelete: ImageButton = mView.findViewById<View>(R.id.imgDelete) as ImageButton
        var mItem: ViewImageData? = null
    }
}
